### PR TITLE
Remove exclusion on map nodes

### PR DESCRIPTION
**Pull request by peicodes**
This PR removes exclusions for map nodes. It also highlights some bugs we didn't know existed.

Follow ups from findings while doing this:
- https://app.shortcut.com/vellum/story/5666/fix-codegen-for-map-node
- https://app.shortcut.com/vellum/story/5667/fix-output-display-reference-on-map-nodes

### CODE REVIEW

**Comment by peicodes** [comment]
@dvargas92495 The test failure with failing to infer types only happens in CI and not local. Could you help me with it? Not sure why it's happening.

File: ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/__init__.py [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 
 class MapNodeDisplay(BaseMapNodeDisplay[MapNode]):
     label = "Map Node"
     node_id = UUID("72cb9f1e-aedd-47af-861e-4f38d27053b6")
     target_handle_id = UUID("0d15cb2c-256e-423e-a489-c9f87e181280")
-    workflow_input_ids_by_name = {}
     node_input_ids_by_name = {"items": UUID("b8d66997-444e-4409-b315-5bef0c06192a")}
     port_displays = {MapNode.Ports.default: PortDisplayOverrides(id=UUID("239a1483-e4f5-4650-81a4-21c77d72cc5e"))}
     display_data = NodeDisplayData(position=NodeDisplayPosition(x=254, y=0), width=None, height=None)
+    output_display = {
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/__init__.py:19]: Until we figure out https://app.shortcut.com/vellum/story/5667/fix-output-display-reference-on-map-nodes, we need to explicitly override this so terminal nodes that refer to the output of a map node uses the correct ID

File: ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/workflow.py [hunk 2]
```diff
@@ -29,10 +29,9 @@ class MapNodeWorkflowDisplay(VellumWorkflowDisplay[MapNodeWorkflow]):
         ),
     )
     inputs_display = {
-        Inputs.items: WorkflowInputsVellumDisplayOverrides(id=UUID("cdc4468f-45e7-46ce-bbe7-d1aa9ad86514")),
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/workflow.py:32]: This is a duplicate and also incorrect ID

File: ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/workflow.py [hunk 3]
```diff
@@ -29,10 +29,9 @@ class MapNodeWorkflowDisplay(VellumWorkflowDisplay[MapNodeWorkflow]):
...
         Inputs.items: WorkflowInputsVellumDisplayOverrides(id=UUID("b8d66997-444e-4409-b315-5bef0c06192a")),
         Inputs.item: WorkflowInputsVellumDisplayOverrides(id=UUID("2619e147-870f-40ec-8f21-f3e131fcd65a")),
-        Inputs.index: WorkflowInputsVellumDisplayOverrides(id=UUID("edecf894-c35b-485a-998f-118833a4b045")),
+        Inputs.index: WorkflowInputsVellumDisplayOverrides(id=UUID("edecf894-c35b-485a-998f-118833a4b045"))
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/display/nodes/map_node/workflow.py:34]: The comma doesn't matter. Feel free to ignore.

File: ee/codegen_integration/fixtures/simple_map_node/code/display/workflow.py [hunk 4]
```diff
@@ -30,9 +30,6 @@ class WorkflowDisplay(VellumWorkflowDisplay[Workflow]):
     )
     inputs_display = {
         Inputs.items: WorkflowInputsVellumDisplayOverrides(id=UUID("cdc4468f-45e7-46ce-bbe7-d1aa9ad86514")),
-        Inputs.items: WorkflowInputsVellumDisplayOverrides(id=UUID("b8d66997-444e-4409-b315-5bef0c06192a")),
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/display/workflow.py:33]: These don't belong here. There is a difference between the Map Node's inputs, and the underlying Subworkflow's inputs. 

We're likely putting it all together in both places in codegen, hence the duplicated `items` and there being 4 inputs in both places.

File: ee/codegen_integration/fixtures/simple_map_node/code/nodes/map_node/__init__.py [hunk 5]
```diff
@@ -7,6 +7,7 @@
 class MapNode(BaseMapNode):
     items = Inputs.items
     subworkflow = MapNodeWorkflow
+    concurrency = 4
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/nodes/map_node/__init__.py:10]: We're not codegenning concurrency overrides

File: ee/codegen_integration/fixtures/simple_map_node/code/nodes/map_node/nodes/search_node.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 from vellum import SearchFiltersRequest, SearchRequestOptionsRequest, SearchResultMergingRequest, SearchWeightsRequest
 from vellum.workflows.nodes.displayable import SearchNode as BaseSearchNode
 
-from ....inputs import Inputs
+from ....nodes.map_node.inputs import Inputs
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/code/nodes/map_node/nodes/search_node.py:4]: This file is inside the subworkflow. We should be using the `Inputs` from the subworkflow, NOT the outer workflow.

File: ee/codegen_integration/fixtures/simple_map_node/display_data/simple_map_node.json [hunk 7]
```diff
@@ -25,6 +26,24 @@
           "variant": "INLINE",
           "workflow_raw_data": {
             "nodes": [
+              {
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/display_data/simple_map_node.json:29]: This is just an order change. Probably doesn't matter, I can switch it back if we prefer.

File: ee/codegen_integration/fixtures/simple_map_node/display_data/simple_map_node.json [hunk 8]
```diff
@@ -400,38 +406,23 @@
...
       "default": null,
-      "extensions": null
+      "extensions": { "color": null }
     }
   ],
   "output_variables": [
     {
       "id": "d9269719-a7a2-4388-9b85-73e329a78d16",
-      "key": "items",
-      "type": "JSON",
```
Comment (peicodes) [inline ee/codegen_integration/fixtures/simple_map_node/display_data/simple_map_node.json:431]: Outputs don't have required

**Review by NgoHarrison — DISMISSED** [review]
All of the changes do seem like codegen so approving and will fix them in the followups on codegen side

**Review by noanflaherty — APPROVED** [review]

**Review by dvargas92495 — APPROVED** [review]

**Pull request merged**